.goreleaser.yml

```diff
@@ -0,0 +1,113 @@
+env:
+- GO111MODULE=on
+- CGO_ENABLED=0
+before:
+  hooks:
+    - go mod tidy
+    - go mod vendor
+builds:
+  - id: olm
+    main: ./cmd/olm
+    binary: olm
+    goos:
+    - linux
+    goarch:
+    - amd64
+    - arm64
+    tags:
+    - json1
+    flags:
+    - -mod=vendor
+    ldflags:
+    - -X {{ .Env.PKG }}/pkg/version.GitCommit={{ .FullCommit }}
+    - -X {{ .Env.PKG }}/pkg/version.OLMVersion={{ .Tag }}
+  - id: catalog
+    main: ./cmd/catalog
+    binary: catalog
+    goos:
+    - linux
+    goarch:
+    - amd64
+    - arm64
+    tags:
+    - json1
+    flags:
+    - -mod=vendor
+    ldflags:
+    - -X {{ .Env.PKG }}/pkg/version.GitCommit={{ .FullCommit }}
+    - -X {{ .Env.PKG }}/pkg/version.OLMVersion={{ .Tag }}
+  - id: cpb
+    main: ./util/cpb
+    binary: cpb
+    goos:
+    - linux
+    goarch:
+    - amd64
+    - arm64
+    tags:
+    - json1
+    flags:
+    - -mod=vendor
+    ldflags:
+    - -X {{ .Env.PKG }}/pkg/version.GitCommit={{ .FullCommit }}
+    - -X {{ .Env.PKG }}/pkg/version.OLMVersion={{ .Tag }}
+  - id: package-server
+    main: ./cmd/package-server
+    binary: package-server
+    goos:
+    - linux
+    goarch:
+    - amd64
+    - arm64
+    tags:
+    - json1
+    flags:
+    - -mod=vendor
+    ldflags:
+    - -X {{ .Env.PKG }}/pkg/version.GitCommit={{ .FullCommit }}
+    - -X {{ .Env.PKG }}/pkg/version.OLMVersion={{ .Tag }}
+dockers:
+- image_templates:
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-amd64
+  dockerfile: Dockerfile.goreleaser
+  use: buildx
+  goos: linux
+  goarch: amd64
+  build_flag_templates:
+  - --platform=linux/amd64
+- image_templates:
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-arm64
+  dockerfile: Dockerfile.goreleaser
+  use: buildx
+  goos: linux
+  goarch: arm64
+  build_flag_templates:
+  - --platform=linux/arm64
+docker_manifests:
+- name_template: quay.io/{{ .Env.IMAGE_REPO }}:v{{ .Major }}.{{ .Minor }}
+  image_templates:
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-amd64
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-arm64
+- name_template: quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}
+  image_templates:
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-amd64
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-arm64
+
+- name_template: quay.io/{{ .Env.IMAGE_REPO }}:latest
+  image_templates:
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-amd64
+  - quay.io/{{ .Env.IMAGE_REPO }}:{{ .Tag }}-arm64
+  create_flags:
+  - --amend
```

Comment:
This produces manifestlist images for v.<minor>.<patch>, v.<minor>, and latest which seems like the right behavior since OLM isn't v1 and thus a major-only tag seems strange.

Something to note, is that docker images are pushed with amd64/arm64 suffixes using a v.<minor>.<patch> format (e.g. whatever tag triggered this action) so I'm not sure whether we're comfortable with that format.

Reply:
We have to be careful with the way `latest` works here. If we push tag v0.21.0 and then later push tag v0.20.1, `latest` will point to v0.20.1 when we probably want it to stay on v0.21.0.

operator-registry solves this with some clever envvar value empty string shenanigans.

Reply:
I think I'm leaning towards omitting the handlement around latest images entirely for the time being and trying to tackle that in a follow-up while we still work out what we'd like to see for tag structure in this repository. WDYT?